dockerfiles/Dockerfile-debian

```diff
@@ -0,0 +1,79 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+FROM debian:bullseye-slim AS build
+
+LABEL org.opencontainers.image.authors="dev@qpid.apache.org"
+
+# UPGRADE: proton > 0.37.0: replace `python3/dist-packages` with `python3.9/site-packages/`
+ARG proton_version=0.37.0
+ARG dispatch_version=1.19.0
+ARG enable_console=OFF
+
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends build-essential cmake pkg-config git ninja-build ca-certificates \
+        libssl-dev libsasl2-dev libuv1-dev swig python3-dev libwebsockets-dev npm && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN git clone --depth 1 -b ${dispatch_version} https://gitbox.apache.org/repos/asf/qpid-dispatch.git && \
+	git clone --depth 1 -b ${proton_version} https://gitbox.apache.org/repos/asf/qpid-proton.git
+
+# build and install into system for laster usage by dispatch, but also install discretely so that it's easier to copy
+RUN mkdir /qpid-proton/build /qpid-proton/install-prefix && cd /qpid-proton/build && \
+	cmake .. -GNinja -DPROACTOR=libuv -DSYSINSTALL_BINDINGS=ON -DCMAKE_INSTALL_PREFIX=/usr -DSYSINSTALL_PYTHON=ON && \
+	cmake --build . --target install && \
+	cmake --install . --prefix /qpid-proton/install-prefix
+
+RUN mkdir /qpid-dispatch/build /qpid-dispatch/install-prefix && cd /qpid-dispatch/build && \
+	cmake .. -GNinja -DCMAKE_INSTALL_PREFIX=/usr -DUSE_VALGRIND=NO -DCONSOLE_INSTALL=${enable_console} && \
```

Comment:
Is Ninja really such a big win? I do use it interactively, but mostly because it is fast on rebuilds. Using make with `-j` option for first build is IMO quite comparable in speed with ninja.

I'm fine with ninja in that it avoids having to mess with the `-j` make option.

Reply:
Yes, that was exactly the reason to use it - to avoid hard coding or messing with `-j`